### PR TITLE
TST/FIX/CLN: Add missing tests smoke for plotting functions

### PR DESCRIPTION
**Pull request by bashtage**
Add missing tests for plotting functions
Fix small bugs in plotting functions
Refactor factorplots to remove bad practices

### CODE REVIEW

File: statsmodels/graphics/regressionplots.py [hunk 1]
```diff
@@ -822,8 +822,6 @@ def plot_leverage_resid2(results, alpha=.05, label_kwargs={}, ax=None,
     alpha : float
         Specifies the cut-off for large-standardized residuals. Residuals
         are assumed to be distributed N(0, 1) with alpha=alpha.
-    label_kwargs : dict
-        The keywords to pass to annotate for the labels.
```
Comment (josef-pkt) [inline statsmodels/graphics/regressionplots.py]: my first response in cases like this is to ask whether there is a missing extension that was planned but never finished. Based on the PR #558 it looks like only the partial regression plot got working label_kwargs.
I don't really know what the annotate options are doing.

Comment (bashtage) [inline statsmodels/graphics/regressionplots.py]: I didn't see anything obvious and I could read the person who wrote it's mind.  Someone can always come back in and enhance it.

**Comment by bashtage** [comment]
This should be put into 0.8 since it fixes bus that show up if someone tries to run the plot code.

**Comment by josef-pkt** [comment]
I'll go over it tomorrow again. should be good to merge and backport

**Comment by josef-pkt** [comment]
@bashtage  Thanks for cleaning this up, code looks much better

merging

**Pull request merged**